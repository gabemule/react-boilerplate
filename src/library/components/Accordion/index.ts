import DsAccordion from "./Accordion"
import { DsAccordionItemProps }from "@/library/types/types";

export { DsAccordion }

export type { DsAccordionItemProps }
