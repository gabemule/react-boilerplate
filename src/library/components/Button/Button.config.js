var IconConfig = require('../Icon/Icon.config')

module.exports = {
  name: 'Button',
  class: 'ds-button',
  props: {
    type: {
      type: String,
      default: 'primary',
      options: ['primary', 'secondary', 'danger', 'warning']
    },
    size: {
      type: String,
      default: 'medium',
      options: ['large', 'medium', 'small']
    },
    disabled: {
      type: Boolean,
      default: false
    },
    icon: {
      type: String,
      default: IconConfig.props.image.default,
      options: IconConfig.props.image.options
    }
  }
}
