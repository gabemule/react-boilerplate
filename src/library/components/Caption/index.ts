import DsCaption from "./Caption";
import { DsCaptionProps } from "./Caption"

export { DsCaption };
export type { DsCaptionProps }
