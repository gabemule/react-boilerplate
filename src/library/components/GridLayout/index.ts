import DsGridLayout from './GridLayout'
import DsGridElement from './GridElement'
import { DsGridLayoutProps } from './GridLayout'

export { DsGridLayout, DsGridElement }
export type { DsGridLayoutProps }
