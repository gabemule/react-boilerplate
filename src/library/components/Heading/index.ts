import DsHeading from "./Heading";
import { DsHeadingProps }from "./Heading";

export { DsHeading };
export type { DsHeadingProps }
