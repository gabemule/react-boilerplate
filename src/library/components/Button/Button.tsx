import React, { HTMLAttributes } from "react";
// import withCustomProps from '@/library/decorators/withCustomProps'
import PropTypes from 'prop-types';

// config
import ButtonConfig from "./Button.config";

// components
import { DsIcon } from "@/library/components/Icon";
// import Button from "@/app/views/components/Button";

export interface DsButtonProps extends HTMLAttributes<HTMLButtonElement> {
  children: string,
  className?: string,
  size?: string,
  disabled?: boolean,
  icon?: string,
  type?: string,
}

const DsButton = ({
  children,
  className,
  onClick,
  size,
  disabled,
  icon,
  type,
}: DsButtonProps) => {
  const buttonSize = size && ButtonConfig.props.size.options.find((value: string) => value === size) ? `${ButtonConfig.class}--${size}` : `${ButtonConfig.class}__huge`;
  const buttonDisabled = disabled && `${ButtonConfig.class}--disabled`;
  const buttonType = type && `${ButtonConfig.class}--${type}`

  return (
    <button
      className={[ButtonConfig.class, buttonSize, buttonType, buttonDisabled, className && className].filter(classes => classes).join(" ")}
      onClick={!disabled ? onClick : () => { }}
      data-testid="button"
    >
      <span className="ds-button__text">
        {children}
        {icon &&
          <span className="ds-button__icon">
            <DsIcon image={icon} color={type === "danger" ? "neutral-00" : ""} />
          </span>
        }
      </span>
    </button>
  );
};

DsButton.propTypes = {
  type: PropTypes.oneOf(ButtonConfig.props.type.options),
  size: PropTypes.oneOf(ButtonConfig.props.size.options),
  disabled: PropTypes.bool,
  icon: PropTypes.oneOf(ButtonConfig.props.icon.options)
};

DsButton.defaultProps = {
  type: ButtonConfig.props.type.default,
  size: ButtonConfig.props.size.default,
  disabled: ButtonConfig.props.disabled.default,
  icon: ButtonConfig.props.icon.default
};

export default DsButton;
