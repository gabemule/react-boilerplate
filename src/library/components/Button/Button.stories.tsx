import React from "react";
import { action } from "@storybook/addon-actions";
import { Meta, Story } from "@storybook/react";

import ButtonConfig from "./Button.config";
import IconConfig from "../Icon/Icon.config";

import { ThemeProvider, ThemeProviderProps } from "@/library/components/ThemeProvider";
import { DsButton, DsButtonProps } from "@/library/components/Button";

const componentProps: Meta = {
  title: "Interação Básica/Button/Button",
  component: DsButton,
  argTypes: {
    size: {
      name: "Size",
      type: {
        name: "string",
        required: false
      },
      defaultValue: ButtonConfig.props.size.default,
      description: "Define o tamanho do Botão",
      table: {
        category: "Style props",
        type: { summary: "string" },
        defaultValue: { summary: "large" }
      },
      control: { type: "select", options: ButtonConfig.props.size.options }
    },
    disabled: {
      name: "Disabled",
      type: {
        name: "string",
        required: false
      },
      defaultValue: ButtonConfig.props.disabled.default,
      description: "Desabilita o botão",
      table: {
        category: "Style props",
        type: { summary: "Boolean" },
        defaultValue: { summary: false }
      },
      control: { type: "boolean", }
    },
    icon: {
      name: "Icon",
      type: {
        name: "string",
        required: false
      },
      defaultValue: IconConfig.props.image.default,
      description: "Define o ícone que será exibido ao lado do texto",
      table: {
        category: "Text with icon props",
        type: { summary: "string" },
        defaultValue: { summary: "essential/info" }
      },
      options: IconConfig.props.image.options,
      control: { type: "select", }
    },
    type: {
      name: "Type",
      type: {
        name: "string",
        required: false
      },
      defaultValue: ButtonConfig.props.type.default,
      description: "Define o ícone que será exibido ao lado do texto",
      table: {
        category: "Style props",
        type: { summary: "string" },
        defaultValue: { summary: "essential/info" }
      },
      options: ButtonConfig.props.type.options,
      control: { type: "select", }
    }
  }
};

export default componentProps;


const Template: Story<ThemeProviderProps & DsButtonProps> = (args) => (
  <ThemeProvider library={"core"} theme={"light"}>
    <DsButton {...args}> Click here </DsButton>
  </ThemeProvider>
);

const TemplateNoProps: Story<ThemeProviderProps & DsButtonProps> = () => (
  <ThemeProvider library={"core"} theme={"light"}>
    <DsButton> Button with no props </DsButton>
  </ThemeProvider>
);

const defaultProps = {
  onClick: action("clicked"),
  disabled: false,
  size: "large",
}

export const Default = Template.bind({});
Default.args = { ...defaultProps };

export const NoProps = TemplateNoProps.bind({});
NoProps.args = {};
