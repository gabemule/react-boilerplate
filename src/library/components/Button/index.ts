import DsButton from "./Button";
import { DsButtonProps }from "./Button";

export { DsButton };
export type { DsButtonProps }
