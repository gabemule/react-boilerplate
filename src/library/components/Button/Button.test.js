import React from 'react'
import { shallow, render } from 'enzyme'

// components
import { DsButton } from '@/library/components/Button'

// config
import ButtonConfig from './Button.config'

// constants
const ButtonClass = ButtonConfig.class
const ButtonText = 'Button Default Text'

describe('Button Component', () => {
  it('Matches Snapshot', () => {
    const wrapper = render(<DsButton>{ButtonText}</DsButton>)
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    const wrapper = shallow(<DsButton>{ButtonText}</DsButton>)
    expect(wrapper).toBeTruthy()
    expect(wrapper.find(ButtonClass)).toBeTruthy()
  })
})
