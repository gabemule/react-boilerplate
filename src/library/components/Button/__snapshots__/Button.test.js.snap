// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Matches Snapshot 1`] = `
LoadedCheerio {
  "0": <button
    class="ds-button ds-button--medium ds-button--primary"
    data-testid="button"
  >
    <span
      class="ds-button__text"
    >
      Button Default Text
    </span>
  </button>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
